src/modules/dashboard/components/CreatePaymentDialog/CreatePaymentDialogForm.tsx

```diff
@@ -0,0 +1,394 @@
+  },
+  noPermissionFrom: {
+    id:
+      'dashboard.CreatePaymentDialog.CreatePaymentDialogForm.noPermissionFrom',
+    defaultMessage:
+      // eslint-disable-next-line max-len
+      'You do not have the {firstRoleRequired} and {secondRoleRequired} permissions required to take this action.',
+  },
+});
+
+interface Props {
+  cancel: () => void;
+  colony: Colony;
+  subscribedUsers: AnyUser[];
+}
+
+const UserAvatar = HookedUserAvatar({ fetchUser: false });
+
+const supRenderAvatar = (address: string, item: ItemDataType<AnyUser>) => (
+  <UserAvatar address={address} user={item} size="xs" notSet={false} />
+);
+
+const CreatePaymentDialogForm = ({
+  cancel,
+  colony,
+  colony: { colonyAddress, domains, tokens },
+  subscribedUsers,
+  handleSubmit,
+  isSubmitting,
+  isValid,
+  setErrors,
+  status,
+  values,
+}: Props & FormikProps<FormValues>) => {
+  const { formatMessage } = useIntl();
+
+  const { tokenAddress, amount } = values;
+  const fromDomain = values.fromDomain
+    ? parseInt(values.fromDomain, 10)
+    : ROOT_DOMAIN_ID;
+
+  const selectedToken = useMemo(
+    () => tokens.find((token) => token.address === values.tokenAddress),
+    [tokens, values.tokenAddress],
+  );
+
+  const tokenOptions = useMemo(
+    () =>
+      tokens.map(({ address, symbol }) => ({
+        value: address,
+        label: symbol || '???',
+      })),
+    [tokens],
+  );
+
+  const { walletAddress } = useLoggedInUser();
+
+  const fromDomainRoles = useTransformer(getUserRolesForDomain, [
+    colony,
+    walletAddress,
+    fromDomain,
+  ]);
+
+  const domainOptions = useMemo(
+    () =>
+      sortBy(
+        domains.map(({ name, ethDomainId }) => ({
+          value: ethDomainId.toString(),
+          label: name,
+        })),
+        ['value'],
+      ),
+
+    [domains],
+  );
+
+  const [
+    loadTokenBalances,
+    { data: tokenBalancesData },
+  ] = useTokenBalancesForDomainsLazyQuery();
+
+  useEffect(() => {
+    if (tokenAddress) {
+      loadTokenBalances({
+        variables: {
+          colonyAddress,
+          tokenAddresses: [tokenAddress],
+          domainIds: [fromDomain],
+        },
+      });
+    }
+  }, [colonyAddress, tokenAddress, fromDomain, loadTokenBalances]);
+
+  const fromDomainTokenBalance = useMemo(() => {
+    const token =
+      tokenBalancesData &&
+      tokenBalancesData.tokens.find(({ address }) => address === tokenAddress);
+    const from = getBalanceFromToken(token, fromDomain);
+    return from;
+  }, [fromDomain, tokenAddress, tokenBalancesData]);
+
+  useEffect(() => {
+    const errors: {
+      amount?: any;
+    } = {};
+
+    if (!selectedToken || !(amount && amount.length)) {
+      errors.amount = undefined; // silent error
+    } else {
+      const convertedAmount = bigNumberify(
+        moveDecimal(
+          amount,
+          getTokenDecimalsWithFallback(selectedToken.decimals),
+        ),
+      );
+      if (!convertedAmount.eq(0)) {
+        errors.amount = MSG.noAmount;
```

Comment:
Hmm, so if convertedAmount is NOT 0, then noAmount?

Reply:
This is wrong, yeah hahaha Should be the contrary